bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Hash.cs

```diff
@@ -1,15 +1,338 @@
+                out var value);
+            status.ThrowIfError();
+            return value.IsInvalid ? null : new ElementModQ(value);
+        }
+
+        /// <summary>
+        /// Hash together the ElementModP values
+        /// </summary>
+        /// <param name="first">first value for the hash</param>
+        /// <param name="second">second value for the hash</param>
+        public static ElementModQ HashElems(ElementModQ first, ElementModP second)
+        {
+            var status = External.HashElems(first.Handle, second.Handle,
+                out var value);
+            status.ThrowIfError();
+            return value.IsInvalid ? null : new ElementModQ(value);
+        }
+
+        /// <summary>
+        /// Hash together the ElementModP values
+        /// </summary>
+        /// <param name="data">List of ElementModP to hash together</param>
+        public static ElementModQ HashElems(List<ElementModP> data)
+        {
+            var dataPointers = new IntPtr[data.Count];
+            for (var i = 0; i < data.Count; i++)
+            {
+                dataPointers[i] = data[i].Handle.Ptr;
+                // TODO: Do we really want to dispose here?
+                data[i].Dispose();
+            }
+
+            var status = External.HashElems(dataPointers, (ulong)data.Count,
+                out var value);
+            status.ThrowIfError();
+            return value.IsInvalid ? null : new ElementModQ(value);
+        }
+
+        /// <summary>
+        /// Hash together the ICryptoHashableType values
+        /// </summary>
+        public static ElementModQ HashElems(string prefix, params ICryptoHashableType[] data)
+        {
+            // TODO: we should define a clear method for 
+            // passing hashable elements back and forth
+            // between C# and C++ so we don't have to do this
+            var hashed = HashElems(prefix);
+            foreach (var item in data)
+            {
+                if (item is CryptoHashableBase)
+                {
+                    if (item is ElementModP p)
+                    {
+                        using (var combined = HashElems(hashed, p))
+                        {
+                            hashed.Reassign(combined);
+                        }
+
+                    }
+                    else if (item is ElementModQ q)
+                    {
+
+                        using (var combined = HashElems(hashed, q))
+                        {
+                            hashed.Reassign(combined);
+                        }
+                    }
+                }
+                else
+                {
+                    throw new ArgumentException("data must be of type CryptoHashableBase");
+                }
+            }
+            return hashed;
+        }
+
+        // TODO: enforce polymorphism
+        // public static ElementModQ HashElems(string prefix, params ICryptoHashableType[] data)

```

Comment:
we probably dont need to keep the commented code here i will remove it, but the crypto hashable implementation is naive and we do need to handle the object assignment better than is being done above wher it just checks if its elementmodp or q